src/reformatters/noaa/hrrr/forecast_48_hour/template.py

```diff
@@ -57,11 +57,27 @@ def update_template() -> None:
 def derive_coordinates(
     ds: xr.Dataset,
 ) -> dict[str, xr.DataArray | tuple[tuple[str, ...], np.ndarray[Any, Any]]]:
+    # following pygrib methodology for lat/lon calculation
+    # https://github.com/jswhit/pygrib/blob/master/src/pygrib/_pygrib.pyx#L1623-L1644
+    lat_corner = 21.138123
+    lon_corner = 237.280472
+    nx = 1799
+    ny = 1059
+    dx, dy = 3000.0, 3000.0
+
+    pj = pyproj.Proj(ds.rio.crs.to_proj4())
+    x_corner, y_corner = pj(lon_corner, lat_corner)
+    x = x_corner + np.arange(nx) * dx
+    y = y_corner + np.arange(ny) * dy
+    xs, ys = np.meshgrid(x, y)
+    lons, lats = pj(xs, ys, inverse=True)
+    proj_xs, proj_ys = pj(lons, lats)
+
     return {
-        # TODO: Derive these coordinates from an actual model file
-        "latitude": (("y", "x"), np.full((ds["y"].size, ds["x"].size), 0.0)),
-        # TODO: Derive these coordinates from an actual model file
-        "longitude": (("y", "x"), np.full((ds["y"].size, ds["x"].size), 0.0)),
+        "y": (("y",), proj_ys[:, 0]),
+        "x": (("x",), proj_xs[0, :]),
```

Comment:
I was kinda tired when I wrote this - there might be a smoother numpy-fu here.

Reply:
Can the values for these just be `x` and `y`? (or if not, do we understand why not?)

the dim name can also be a single string if there's only one dim, eg

```suggestion
        "y": ("y", y),
        "x": ("x", x),
```

Reply:
See I knew I was overthinking it